src/Interpreters/TreeRewriter.cpp

```diff
@@ -739,10 +739,10 @@ void collectJoinedColumns(TableJoin & analyzed_join, ASTTableJoin & table_join,
             analyzed_join.setStrictness(table_join.strictness);
         }
 
-        /// FIXME, validate if the combination of lag_behind(...) + strictness makes sense.
-        if (analyzed_join.lagBehindInterval() != 0)
-            /// Add lag behind columns
-            data.lagBehindASTToKeys();
+        /// FIXME, validate if the combination of lag_behind(...)/aligned_with(...) + strictness makes sense.
+        if (analyzed_join.requiredJoinAlignment())
+            /// Add alingned with timestamp columns
+            data.timestampASTToKeys();
```

Comment:
we may rename timestamp to `alignedTimestamp`, `aligned_timestamp_key`, `aligned_timestamp_column_name` etc.